lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -901,6 +906,8 @@ selectAssets era (ProtocolParameters pp) utxoAssumptions outs redeemers
                 , txChange = view #skeletonChange skeleton
                 , txPaymentTemplate = view #template <$>
                     assumedInputScriptTemplate utxoAssumptions
+                , txReferenceScript =
```

Comment:
As said in DM: The changes to `Write.{Balance, SizeEstimation, UTxOAssumptions}` shouldn't be needed. If the new inline-script-containing `TxOut` is part of the partial tx given to `balanceTx`, then `balanceTx` will know of its size and fee-contribution.